src/devices/machine/generic_spi_flash.cpp

```diff
@@ -161,7 +161,7 @@ void generic_spi_flash_device::write(uint8_t data)
 	{
 		if (m_spi_state == READY_FOR_READ)
 		{
-			m_spilatch = m_spiptr[(m_spiaddr++) & 0xffffff];
+			m_spilatch = m_spiptr[(m_spiaddr++) & (m_length-1)];
```

Comment:
If you’re going to be using `m_length - 1` as a mask, you need a check somewhere to ensure it’s a power-of-two, otherwise it will behave very strangely.  This is particularly important when the length comes from external data that any user can mess with.

Reply:
it will now reject ROMs that are non power-of-2